### PR TITLE
Make bbp::sonata::Selection comparable

### LINKED ISSUE #61: Make `bbp::sonata::Selection` comparable
Primarily to make unit tests easier, but can be useful for other things as well.

### PR DESCRIPTION
**Pull request by mgeplf**
* Closes #61

### CODE REVIEW

File: include/bbp/sonata/population.h [hunk 1]
```diff
@@ -50,6 +50,9 @@ class SONATA_API Selection
     const Ranges ranges_;
 };
 
+bool SONATA_API operator==(const Selection&, const Selection&);
+bool SONATA_API operator!=(const Selection&, const Selection&);
```
Comment (matz-e) [inline include/bbp/sonata/population.h]: Can we also have `operator bool`?

Comment (tristan0x) [reply include/bbp/sonata/population.h]: It is in the Python bindings, but it feels weird to me to override this operator in C++. One should use the `empty()` method like for STL containers: `if (!my_vector.empty())`.

Comment (mgeplf) [reply include/bbp/sonata/population.h]: we already have `empty()` like in std::vector

Comment (mgeplf) [reply include/bbp/sonata/population.h]: Or what @tristan0x said.

Comment (matz-e) [reply include/bbp/sonata/population.h]: Maybe I just like my C++ code to feel more Pythonic…

Comment (mgeplf) [reply include/bbp/sonata/population.h]: I try and follow the principle of least astonishment, and in the C++ community, my read is that containers have an `empty()` and having a bool operator would be surprising.

Comment (matz-e) [reply include/bbp/sonata/population.h]: …least astonishment…C++… that's a good one. Anyways, fine with me, I don't have strong feelings here.

Comment (tristan0x) [reply include/bbp/sonata/population.h]: Thank you for your comment anyway @matz-e

**Pull request merged**